### DISCUSSION TITLE
ERR GetAppTraces, Jaeger GRPC client error: rpc error: code = Unavailable desc = connection closed

### DISCUSSION THREAD

**Discussion by samene** [kiali/kiali#4193] [General]
**Describe the bug**

Kiali Operator
```
external_services:
  tracing:
    in_cluster_url: http://tracing-query.istio-system:16686/jaeger
```
In the Kiali app when I try to navigate to the traces get ```503``` with below error in pod log.

```
ERR GetAppTraces, Jaeger GRPC client error: rpc error: code = Unavailable desc = connection closed
```

But I am able to open jaeger UI and also do a curl
```
curl http://tracing-query.istio-system:16686/jaeger/api/traces

{"data":null,"total":0,"limit":0,"offset":0,"errors":[{"code":400,"msg":"parameter 'service' is required"}]}
```

**Versions used**
Kiali 1.37.0, Jaeger 1.24.0 and istio 1.10.2
Kubernetes 1.21.1 (kubeadm bare-metal)

**Comment by xeviknal** [comment]
Have you had the chance to go through the [tracing settings](https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml#L615-L642) into the kiali-cr?

Another thing you can try is to disable the [grpc](https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml#L639). It fallbacks to a regular HTTP calls.

**Reply by xeviknal** [reply]
btw, I am converting this into a discussion. I have the feeling that it is not a problem in kiali side. It is more problems in the configuration. I am wrong, we'll for sure make this an issue. 

Thank you anyway for reaching us

**Reply by samene** [reply]
Thanks. The use_grpc: false resolved the issue for me.

**Comment by samene** [comment]
BTW, I see this in your FAQ section
_If for some reason the GRPC connection fails and you think it shouldn’t (e.g. your reverse-proxy supports it, and the non-grpc config works fine), please get in touch with us._

I don't have a reverse-proxy, so not sure why the GRPC connection fails. Is there any logs I can gather to help troubleshoot?

**Reply by xeviknal** [reply]
hey @samene thank you for coming back to us. Indeed, we should investigate further. And as @jshaughn mentioned it might be caused by the jaeger version. Being said that, I've opened an issue for taking care about it: https://github.com/kiali/kiali/issues/4207

**Comment by jshaughn** [comment]
I am seeing the same behavior after switching to Istio 1.11 beta.0.  from 1.10.2.   One difference is that the version of Jaeger in the Istio addon is updated from  v1.20 to v1.22.  @samene , I see you were using Jaeger 1.24, which is even newer that 1.22.  I expect something has changed in Jaeger and @xeviknal  we may need to make a change in Kiali.

**Comment by xeviknal** [comment]
hi @samene,

I think I found the root cause for this problem. Jaeger team has separated ports for gRPC and HTTP:

> Separate Ports for GRPC and HTTP requests in Query Server (#2387, @rjs211) https://github.com/jaegertracing/jaeger/releases/tag/v1.20.0

Jaeger uses port 80 for HTTP traffic and 16685 for gRPC traffic. Therefore, just applying the following config you should have jaeger working back with grpc and without bumping any new kiali version.

```yaml
    tracing:
      in_cluster_url: "http://tracing.istio-system:16685/jaeger"
      use_grpc: true
```

Note the change in the port: 16685.

I did reproduce the error with Jaeger 1.22 and this fixed the problem. Let us know.

**Reply by mkushal** [reply]
@xeviknal , Thanks it worked  . You saved lots of time for me.

**Reply by OneideLuizSchneider** [reply]
@xeviknal thanks \m/
